notebooks/103-tensorflow-training-openvino/103-tensorflow-training-openvino.ipynb

```diff
@@ -0,0 +1,1141 @@
+    "                 --output_dir \"{model_path.parent}\"\n",
+    "                 --model_name \"{ir_model_name}\"\n",
+    "                 \"\"\"\n",
+    "mo_command = \" \".join(mo_command.split())\n",
+    "print(\"Model Optimizer command to convert TensorFlow to OpenVINO:\")\n",
+    "print(mo_command)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run the Model Optimizer (overwrites the older model)\n",
+    "print(\"Exporting TensorFlow model to IR... This may take a few minutes.\")\n",
+    "!$mo_command"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Preprocessing Image Function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def pre_process_image(imagePath, img_height=180):\n",
+    "    # Model input format\n",
+    "    n, c, h, w = [1, 3, img_height, img_height]\n",
+    "    image = Image.open(imagePath)\n",
+    "    image = image.resize((h, w), resample=Image.BILINEAR)\n",
+    "\n",
+    "    # Normalize to keep data between 0 - 1\n",
+    "    image = np.array(image) \n",
+    "\n",
+    "    # Change data layout from HWC to CHW\n",
+    "    image = image.transpose((2, 0, 1))\n",
+    "    input_image = image.reshape((n, c, h, w))\n",
+    "\n",
+    "    return input_image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### OpenVINO Inference Engine Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class_names=['daisy', 'dandelion', 'roses', 'sunflowers', 'tulips']\n",
+    "\n",
+    "model_xml = f'flower_ir.xml'\n",
+    "model_bin = f'flower_ir.bin'\n",
+    "\n",
+    "# Load network to the plugin\n",
+    "ie = IECore()\n",
+    "net = ie.read_network(model=model_xml, weights=model_bin)\n",
+    "\n",
+    "#Neural Compute Stick\n",
+    "#exec_net = ie.load_network(network=net, device_name=\"MYRIAD\")\n",
+    "exec_net = ie.load_network(network=net, device_name=\"CPU\")\n",
+    "\n",
+    "del net\n",
+    "\n",
+    "input_layer = next(iter(exec_net.input_info))\n",
+    "output_layer = next(iter(exec_net.outputs))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Run the Inference Step"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run the Inference on the Input image...\n",
+    "inp_img_url = \"https://upload.wikimedia.org/wikipedia/commons/4/48/A_Close_Up_Photo_of_a_Dandelion.jpg\"\n",
+    "inp_file_name = 'A_Close_Up_Photo_of_a_Dandelion.jpg'\n",
+    "\n",
+    "#Download the image from the storage\n",
+    "output = subprocess.check_output(f\"curl {inp_img_url} --output {inp_file_name}\", shell=True)\n",
```

Comment:
Is curl included on a default Ubuntu 18.04 desktop install? http://releases.ubuntu.com/18.04/ubuntu-18.04.5-desktop-amd64.manifest seems to indicate it isn't. It's safer to use urllib.request.urlretrieve. I can also change this after merge if you want.

Reply:
I think we will do another update on that!